### PR TITLE
Fix issue removing embeds when editing with files and add `embeds` to `PartialMessage.edit`

### PR DESCRIPTION
**Pull request by DenverCoder1**
## Summary

Feature:
* Adds support for sending `embeds` in `PartialMessage.edit`

Fixes:
* Replaces use of deprecated `embed` field in `PartialMessage.edit` with `embeds`
* **Fixes issue when trying to remove content, embeds, or other attributes when editing a message with files**

## Bug fix details

This PR fixes a bug where if you pass `embed=None` or `embeds=[]` to an edit method while also sending files, the embeds do not get removed.

This bug also applies to sending `view=None`, a view with no components, an empty list of stickers, or empty string as content.

Example:

```py
@bot.command(name="pme")
async def partial_message_embeds(ctx: commands.Context):
    pm = ctx.channel.get_partial_message(956263847562477619)
    await pm.edit(content="Test 1", embeds=[nextcord.Embed(title="Test"), nextcord.Embed(title="Test 2")])
    await asyncio.sleep(2)
    # this should remove the embeds
    await pm.edit(content="Test 2", embed=None, files=[nextcord.File("nextcord-logo.png")])
    # this should also remove the embeds
    await pm.edit(content="Test 2", embeds=[], files=[nextcord.File("nextcord-logo.png")])
```

This was fixed by checking for `embeds` being `None` rather than checking `if embeds` (which would skip adding the embeds field if the list is empty.

## Testing

Code for testing changes:

https://paste.nextcord.dev/?id=1648148708552851

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: nextcord/message.py [hunk 1]
```diff
@@ -1850,13 +1857,14 @@ async def edit(self, **fields: Any) -> Optional[Message]:
...
-        except KeyError:
-            pass
-        else:
-            if embed is not None:
-                fields['embed'] = embed.to_dict()
+        if 'embed' in fields and 'embeds' in fields:
+            raise InvalidArgument('Cannot pass both embed and embeds parameter to edit()')
+
+        if 'embed' in fields and fields['embed'] is not None:
+            fields['embed'] = fields['embed'].to_dict()
```
Comment (MaskDuck) [inline nextcord/message.py:1864]:
```suggestion
            fields['embed'] = [fields['embed'].to_dict()]
```
api v10 thing

**Review by MaskDuck — CHANGES_REQUESTED** [review]

File: nextcord/message.py (hunk 1 above)
Comment (DenverCoder1) [reply nextcord/message.py:1864]: The send_multipart_helper converts embed to embeds already so it doesn't need to be handled here

Comment (MaskDuck) [reply nextcord/message.py:1864]: k

**Review by MaskDuck — APPROVED** [review]

File: nextcord/message.py (hunk 1 above)
Comment (DenverCoder1) [reply nextcord/message.py:1864]: Actually, it seems like it only gets converted if files are sent. I'll look into this further  :+1:

File: nextcord/message.py [hunk 2]
```diff
@@ -1850,13 +1857,15 @@ async def edit(self, **fields: Any) -> Optional[Message]:
...
-            pass
-        else:
-            if embed is not None:
-                fields['embed'] = embed.to_dict()
+        if 'embed' in fields and 'embeds' in fields:
+            raise InvalidArgument('Cannot pass both embed and embeds parameter to edit()')
+
+        if 'embed' in fields:
+            embed = fields.pop('embed')
+            fields['embeds'] = [embed] if embed is not None else []
```
Comment (Skelmis) [inline nextcord/message.py:1865]: Do you not required `.to_dict()` on this setup? It appears `embeds` and the previous implementation called it.

**Review by Skelmis — CHANGES_REQUESTED** [review]
Looks like a regression from the change, however, if tested feel free to ignore it.

File: nextcord/message.py (hunk 2 above)
Comment (DenverCoder1) [reply nextcord/message.py:1865]: Yeah, I think I messed that up after a few rewrites. It probably does need to be added back.

**Comment by DenverCoder1** [comment]
@Skelmis Issue is fixed, all cases have been tested again

**Review by Skelmis — APPROVED** [review]

**Review by ooliver1 — APPROVED** [review]

**Pull request merged**